src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -306,10 +392,54 @@ public boolean equals(Object o)
      */
     public static void setRandomProfession(EntityVillager entity, Random rand)
     {
-        Set<String> entries = INSTANCE.professions.getKeys();
+        ArrayList<String> entries = new ArrayList(INSTANCE.professionsByName.keySet());
         int prof = rand.nextInt(entries.size());
-        //TODO: Grab id range from internal registry
-        entity.setProfession(rand.nextInt(5));
+        VillagerProfession profession = INSTANCE.professionsByName.get(entries.get(prof));
+        entity.setProfession(profession.id);
```

Comment:
vanilla manually sets the profession IDs for any villagers which should be specific, then randomly picked any others.  this should just affect that set which is randomly picked, which is why I didn't think it will change anything, besides getting modded villager professions in the random villagers, should that not occur? I think preventing that from occuring will prevent the ability to breed for specific villagers as well.